### ISSUE TITLE
The special value %100% that shows the hint when the student answers the question correctly is not working

### ISSUE THREAD

**Issue by jaguarfi** [apluslms/a-plus-rst-tools#112]
I am not sure what is the use case for the `%100%` character, but I could not find anything related to this in the code source.

**Comment by Mankro** [comment]
The code that uses the option is in the MOOC-Grader questionnaire grading code: https://github.com/apluslms/mooc-grader/blob/42f1a810c0c19b07ebefe73cce73ef182a99cf47/access/types/forms.py#L606

**Comment by jaguarfi** [comment]
Yes, but I assume that the a-plus-rst-tools must verify that the `%100%` character is in use. or at least define a default value for the feedback value

**Comment by jaguarfi** [comment]
I also assume that the `!` and the `%100%` characters are somehow bounded. I was expecting to prepend the `!` character to the hint in for giving feedback when the right answer was not selected, and then, use the `%100%` character to give feedback when the correct answer was selected. Is that the use case we want. or how the `%100%` symbol is supposed to work?
```rst
...
!b § option **b** must be selected
%100%b § option **b** is correct
```

**Comment by Mankro** [comment]
> Yes, but I assume that the a-plus-rst-tools must verify that the `%100%` character is in use, or at least define a default value for the feedback value.
> 
> https://github.com/apluslms/a-plus-rst-tools/blob/135b5c97dc948ffbc87274777ab209fa0e057a33/directives/questionnaire.py#L293

I don't understand what kind of default value you expect to see. A-plus-rst-tools saves the `%100%` key in the config.yaml and the MOOC-Grader uses it to show the hint if the question is answered correctly. In a checkbox question, the correct answer may require selecting several checkboxes. `%100%` shows the hint when the question is answered correctly.

You can't combine other characters to the `%100%` value. Your example of appending `b` like `%100%b` will not work. You only use `%100%` on the left side of `§` in the RST.

I see you misunderstood the `%100%` first, but now it should be clear. I can now also see that it is not working in A+ v1.8. It worked in v1.7, but the changes to hints in v1.8 have broken this special value. This bug should be fixed.

You can test this question in A+ v1.7 and v1.8:

```
.. questionnaire:: questionnaire_demo 10
  :title: A simple multiple-choice questionnaire

  .. pick-any:: 10
    :required:

    Subdirective ``pick-any`` defines a multiple-choice question.

    When :math:`(x + 1)^2 = 16`, what is :math:`x`?

    a. 4
    *b. an integer
    *c. 3
    d. an irrational number
    ?e. -3
    *f. -5

    %100% § You got full points and this hint.
```

**Comment by jaguarfi** [comment]
Now, with your explanation, it makes more sense. Maybe we need to edit the README.md file. I attach some screenshots on what the user should expect.

*Another test case*
```rst
.. questionnaire:: required-answers-and-show-model 30
  :title: Questionnaire with 100% feedback, answers required and show-model setting enabled.
  :submissions: 5
  :show-model: true

  This questionnaire required the students to answer the questions. The questionnaire also shows the model
  answer to the students after the exercise deadline has passed. If the students answer the single-choice and
  the multiple-choices question correctly, they will see a hint indicating that the answer is 100% correct.

  .. pick-one:: 10
    :required:

    **Modifiers used in this question:** ``required``.

    When :math:`(x + 1)^3 = 27`, what is :math:`x`?

    a. 9
    *b. 2
    c. 3

    %100% § The answer is 100% correct
    a § Not quite. Remember the cube root.
    b § Correct!
    c § Rather close. Remember that you can add or subtract the same number to both sides of the equation.

  .. pick-any:: 10
    :required:

    **Modifiers used in this question:** ``required``.

    When :math:`(x + 1)^2 = 16`, what is :math:`x`?

    a. 4
    *b. an integer
    *c. 3
    d. an irrational number
    e. -3
    *f. -5

    %100% § The answers are 100% correct

  .. freetext:: 10
    :required:

    **Modifiers used in this question:** ``required``.

    Type the word ``mandatory`` in the following text box.

    mandatory
```

*Hint* `%100%` ***was working*** *in v1.7*
![questionnaire_feedback_working](https://user-images.githubusercontent.com/10437475/107634687-9b269280-6c72-11eb-8496-bbf724f37f91.png)

*Hint* `%100%` ***is not working*** *in v1.8*
![questionnaire-feedback-stopworking](https://user-images.githubusercontent.com/10437475/107634698-9f52b000-6c72-11eb-9250-7dc2299bd10e.png)
